public/Add-DbaReplArticle.ps1

```diff
@@ -0,0 +1,179 @@
+        This avoids overwhelming you with "sea of red" exceptions, but is inconvenient because it basically disables advanced scripting.
+        Using this switch turns this "nice by default" feature off and enables you to catch exceptions with your own try/catch.
+
+    .PARAMETER WhatIf
+        If this switch is enabled, no actions are performed but informational messages will be displayed that explain what would happen if the command were to run.
+
+    .PARAMETER Confirm
+        If this switch is enabled, you will be prompted for confirmation before executing any operations that change state.
+
+    .NOTES
+        Tags: Replication
+        Author: Jess Pomfret (@jpomfret), jesspomfret.com
+
+        Website: https://dbatools.io
+        Copyright: (c) 2023 by dbatools, licensed under MIT
+        License: MIT https://opensource.org/licenses/MIT
+
+        https://learn.microsoft.com/en-us/sql/relational-databases/replication/publish/define-an-article?view=sql-server-ver16#RMOProcedure
+
+    .LINK
+        https://dbatools.io/Add-DbaReplArticle
+
+    .EXAMPLE
+        PS C:\> Add-DbaReplArticle -SqlInstance mssql1 -Database Northwind -Publication PubFromPosh -Name TableToRepl
+
+        Adds the TableToRepl table to the PubFromPosh publication from mssql1.Northwind
+
+
+    .EXAMPLE
+        PS C:\> $article = @{
+                    SqlInstance           = "mssql1"
+                    Database              = "pubs"
+                    Publication           = "testPub"
+                    Name                  = "publishers"
+                    Filter                = "city = 'seattle'"
+                }
+        PS C:\> Add-DbaReplArticle @article -EnableException
+
+        Adds the publishers table to the TestPub publication from mssql1.Pubs with a horizontal filter of only rows where city = 'seattle.
+
+    .EXAMPLE
+        PS C:\> $cso = New-DbaReplCreationScriptOptions -Options NonClusteredIndexes, Statistics
+        PS C:\> $article = @{
+                    SqlInstance           = 'mssql1'
+                    Database              = 'pubs'
+                    Publication           = 'testPub'
+                    Name                  = 'stores'
+                    CreationScriptOptions = $cso
+                }
+        PS C:\> Add-DbaReplArticle @article -EnableException
+
+        Adds the stores table to the testPub publication from mssql1.pubs with the NonClusteredIndexes and Statistics options set
+        includes default options.
+    #>
+    [CmdletBinding(DefaultParameterSetName = "Default", SupportsShouldProcess, ConfirmImpact = 'Medium')]
+    param (
+        [parameter(Mandatory, ValueFromPipeline)]
+        [DbaInstanceParameter[]]$SqlInstance,
+        [PSCredential]$SqlCredential,
+        [parameter(Mandatory)]
+        [String]$Database,
+        [parameter(Mandatory)]
+        [String]$Publication,
```

Comment:
```suggestion
        [string]$Publication,
```

Reply:
Are we also going to support piping to this command from say `Get-DbaReplPublication`?